### PR TITLE
Add gofmt to CI

### PR DESCRIPTION
**Pull request by omorsi**
This PR adds a check for the code format in the CI github action.

### CODE REVIEW

**Comment by omorsi** [comment]
@pavolmarko

**Review by groob — APPROVED** [review]

**Comment by omorsi** [comment]
I have changed the job name after your approval, PTAL :)

**Comment by omorsi** [comment]
Will run the formatting job only on one platform, I think this is sufficient. Please tell me if that's wrong.

**Comment by groob** [comment]
> Will run the formatting job only on one platform, I think this is sufficient. Please tell me if that's wrong.

I think that's fine.

File: .github/workflows/CI.yml
Comment (groob) [inline .github/workflows/CI.yml:32]: I'd use ubuntu. it's easier to provision linux containers than windows and macos, which I would think would mean it runs faster/less issues.

Comment (omorsi) [reply .github/workflows/CI.yml]: Done, but just to make sure I understand, according to https://resources.github.com/whitepapers/GitHub-Actions-Cheat-sheet/#:~:text=A%20list%20of%20the%20steps%20that,all%20on%20the%20same%20virtual%20environment. It seams that the VM setup will happen only once per other matrix dimensions for MacOS, right?

Comment (groob) [reply .github/workflows/CI.yml]: /shrug 
I'm making assumptions based on my experience that setting up linux is easier than macOS. I could be wrong when it comes to github actions specifically.

Comment (omorsi) [reply .github/workflows/CI.yml]: oh sorry for not being clear.
I meant that the difference between conditionally running the Format on Linux instead of MacOS is how fast will the gofmt command run on Linux vs MacOS and not on how fast setting a Linux VM vs setting MacOS VM because they will be set regardless for the previous steps, right?

**Pull request merged**
